### ISSUE TITLE
Change `RiverpodComponentMixin.onLoad` to return `FutureOr<void>`

### ISSUE THREAD

**Issue by tibotix** [flame-engine/flame#2961]
# What could be improved

The function signature of `RiverpodComponentMixin.onLoad` could be changed from `void onLoad()` to `FutureOr<void> onLoad`.

# Why should this be improved

For example, when trying to mixin `RiverpodComponentMixin` to a `TextBoxComponent`, a function signature mismatch occurs, as `RiverpodComponentMixin.onLoad` has signature `void onLoad` and `TextBoxComponent.onLoad` has signature `Future<void> onLoad`.
I know i could just make a separate _riverpod aware_ component that extends `Component` (which in fact has the signature `FutureOr<void> onLoad`) (such as in the [documentation](https://docs.flame-engine.org/latest/bridge_packages/flame_riverpod/component.html#riverpodcomponentmixin)), but i would like to have the provider logic inside the actual component. 
Here is some code to see what i am talking about.
```dart
class MyTextBox extends TextBoxComponent with RiverpodComponentMixin { // ERROR: can't mixin  (signature mismatch)
  @override
  void onMount() {
    addToGameWidgetBuild(() {
      ref.listen(myProvider, (previous, next) {
        // do some things such as updating text contents
      });
    });
    super.onMount();
  }
}
```

# Any risks?

I don't know if having the `RiverpodComponentMixin` attached to an actual component other than the plain `Component` class poses any problems, but i don't think so.
Theoretically this is an API change, since the signature changes. However, this signature change will not break any code, because the way `FutureOr<T>` is defined.

# More information

i could do a PR if this is considered useful. But for such a small change i wouldn't be bothered if you would change it yourself real quick.

**Comment by markvideon** [comment]
> @markvideon

Seems like a totally sensible improvement to me!

**Comment by spydon** [comment]
Released in flame_riverpod v5.1.5
